crates/chain-config/Cargo.toml

```diff
@@ -14,18 +14,20 @@ description = "Fuel Chain config types"
 anyhow = { workspace = true }
 bech32 = "0.9.0"
 fuel-core-storage = { workspace = true }
-fuel-core-types = { workspace = true, features = [
-    "serde",
-    "random",
-] }
+fuel-core-types = { workspace = true, default-features = false, features = ["random", "serde"] }
```

Comment:
Why do we need `random` here?

Reply:
I will check to see what dependency/dependencies used it. If I can remove it, I will do so here.

Reply:
It looks like `ChainConfig::local_testnet()` uses `SecretKey::random(&mut rng);`, and `SecretKey::random(..)` is behind the `random` feature. 
Instead of requiring `random`, I can add the `random` feature to `chain-config`. Then:
- we put `ChainConfig::local_testnet()` behind the `random` feature
- we only enable `fuel-core-types:random` when we use `chain-config:random`

Reply:
After some digging, it looks like it is necessary to include `fuel-core-types:random` because `random` allows us to generate valid `SecretKey` values. Since we need to support `fuel-core-chain-config --no-default-features`, we cannot put `fuel-core-types:random` behind another feature flag: Creating keys without `random`, and using something like `SecretKey::default` instead, creates invalid keys in `local_testnet` and tests (maybe `default` should be removed, since default keys are invalid).  
The simplest approach is to keep the `random` feature in `fuel-core-types`.

Reply:
As we discussed during our huddle, we hardcoded the `SecretKeys` in `local_testnet` (based on printing secrets to console using the existing seed) allowing us to remove the `random` feature flag. This removed our transitive dependency on `getrandom`, allowing us to successfully compile `fuel-core-chain-config` on WASM32. Lastly, I also added a new `random_testnet` function that depends on the `random` flag (thus not available in WASM32 land) if users want to have true random secret keys.